ludwig/features/sequence_feature.py

```diff
@@ -270,19 +261,17 @@ def add_feature_data(
 
 @register_input_feature(SEQUENCE)
 class SequenceInputFeature(SequenceFeatureMixin, InputFeature):
-    encoder = "parallel_cnn"
-    max_sequence_length = None
-
-    def __init__(self, feature, encoder_obj=None):
-        super().__init__(feature)
+    def __init__(self, input_feature_config: Union[SequenceInputFeatureConfig, Dict], encoder_obj=None, **kwargs):
+        input_feature_config = self.load_config(input_feature_config)
+        super().__init__(input_feature_config, **kwargs)
         # TODO: Potentially abstract this feature-specific attribute overwrite to a consolidated design.
-        if "vocab" in feature:
-            feature["vocab_size"] = len(feature["vocab"])
-        self.overwrite_defaults(feature)
+        self.encoder_config = input_feature_config.encoder
+        if getattr(self.encoder_config, "vocab", None):
+            self.encoder_config.vocab_size = len(self.encoder_config.vocab)
```

Comment:
we can revise this later, but I'm not a fan of having two parameters for vocab and vocab size around

Reply:
Agreed it is a little clunky and should probably be set more elegantly. I will create an issue on my Linear board to refactor this in a followup PR.